src/common/list-syntax.k

```diff
@@ -0,0 +1,311 @@
+syntax TypeParamList ::= List{TypeParam,","} [klabel('TypeParamList)]
+syntax AbstractMethodModList ::= List{AbstractMethodMod, ""} [klabel('AbstractMethodModList)]
+syntax AnnoAbstractMethodModList ::= List{AnnoAbstractMethodMod, ""} [klabel('AnnoAbstractMethodModList)]
+syntax AnnoMethodModList ::= List{AnnoMethodMod, ""} [klabel('AnnoMethodModList)]
+syntax AnnoVarModList ::= List{AnnoVarMod, ""} [klabel('AnnoVarModList)]
+syntax AnnoClassModList ::= List{AnnoClassMod, ""} [klabel('AnnoClassModList)]
+syntax AnnoConstrModList ::= List{AnnoConstrMod, ""} [klabel('AnnoConstrModList)]
+syntax AnnoConstantModList ::= List{AnnoConstantMod, ""} [klabel('AnnoConstantModList)]
+syntax AnnoFieldModList ::= List{AnnoFieldMod, ""} [klabel('AnnoFieldModList)]
+syntax AnnoInterfaceModList ::= List{AnnoInterfaceMod, ""} [klabel('AnnoInterfaceModList)]
+syntax AnnoList ::= List{Anno, ""} [klabel('AnnoList)]
+syntax AnnoElemDecList ::= List{AnnoElemDec, ""} [klabel('AnnoElemDecList)]
+syntax InterfaceMemberDecList ::= List{InterfaceMemberDec, ""} [klabel('InterfaceMemberDecList)]
+syntax ElemValPairList ::= List{ElemValPair,","} [klabel('ElemValPairList)]
+syntax ElemValList ::= List{ElemVal,","} [klabel('ElemValList)]
+syntax StringPartList ::= List{StringPart,""} [klabel('StringPartList)]
+syntax EnumConstList ::= List{ EnumConst, ","} [klabel(' EnumConstList)]
+syntax ClassBodyDecList ::= List{ClassBodyDec, ""} [klabel('ClassBodyDecList)]
+syntax BlockStmList ::= List{BlockStmt, ""} [klabel('BlockStmList)]
+syntax SwitchGroupList ::= List{SwitchGroup, ""} [klabel('SwitchGroupList)]
+syntax SwitchLabelList ::= List{SwitchLabel, ""} [klabel('SwitchLabelList)]
+
+
+ syntax UserList ::=   BlockStmList
+                      | SwitchGroupList
+                      | SwitchLabelList
+                      | InterfaceTypeList
+                      | ExceptionTypeList
+                      | IdList
+                      | TypeDecList
+                      | VarDecList
+                      | ImportDecList
+                      | ActualTypeArgList
+                      | ClassOrInterfaceTypeList
+                      | TypeParamList
+                      | AbstractMethodModList
+                      | AnnoAbstractMethodModList
+                      | AnnoMethodModList
+                      | AnnoVarModList
+                      | AnnoClassModList
+                      | AnnoConstrModList
+                      | AnnoConstantModList
+                      | AnnoFieldModList
+                      | AnnoInterfaceModList
+                      | AnnoList
+                      | AnnoElemDecList
+                      | InterfaceMemberDecList
+                      | ElemValPairList
+                      | ElemValList
+                      | StringPartList
+                      | EnumConstList
+                      | ClassBodyDecList
+
+
+                      syntax   InterfaceType
```

Comment:
For XXXList, my mistake. For XXX, They are just put there temporarily, I plan to put list together for the ease of management, but what within list should be put in their modules, currently they are not defined, when they are perfectly defined, they should not be here in this module anyway.

Shijiao

-----Original Message-----
From: "Grigore Rosu" <notifications@github.com>
Sent: ‎1/‎1/‎2015 15:57
To: "kframework/java-semantics" <java-semantics@noreply.github.com>
Cc: "laurayuwen" <laurayuwen@gmail.com>
Subject: Re: [java-semantics] List test (#19)

In src/common/list-syntax.k:
> +                      | AnnoClassModList
> +                      | AnnoConstrModList
> +                      | AnnoConstantModList
> +                      | AnnoFieldModList
> +                      | AnnoInterfaceModList
> +                      | AnnoList
> +                      | AnnoElemDecList
> +                      | InterfaceMemberDecList
> +                      | ElemValPairList
> +                      | ElemValList
> +                      | StringPartList
> +                      | EnumConstList
> +                      | ClassBodyDecList
> +
> +
> +                      syntax   InterfaceType
Why are these indented more than the other syntax declarations?
—
Reply to this email directly or view it on GitHub.=